terminal/src/com/jediterm/terminal/ui/JediTermWidget.java

```diff
@@ -214,6 +218,12 @@ public TerminalSession getCurrentSession() {
   @Override
   public TerminalSession createTerminalSession(TtyConnector ttyConnector) {
     setTtyConnector(ttyConnector);
+    new TtyConnectorWaitFor(ttyConnector, Executors.newSingleThreadExecutor()).setTerminationCallback(x -> {
+      for (TerminalWidgetListener listener : myListeners) {
+        listener.allSessionsClosed(this);
```

Comment:
Why does it invoke allSessionClosed here when there is only one session which is closed?

Reply:
Because the single session is closed, hence all are. This is to notify the listeners that there's no active content in the widget anymore, so some hosts may choose to close the widget.

Reply:
Maybe I miss something, but can't a session be closed by a remote end while others remain active?

Reply:
I don't get it. I suppose that JediTermWIdget only supports one session. What I'm not sure about is whether I could just move this code to the EmulatorTask and avoid using the WaitFor.

Reply:
Yes you are right - it supports only one session, I see, and in the tabbed widget, you check that there are 0 tabs, ok. 
About moving it to the EmulatorTasks, it's hard to say whether it is a better place. But surely it's better to avoid using WaitFor as it creates an additional thread and we surely have the information about the closing session in several places.